### ISSUE TITLE
[java] GuardLogStatementJavaUtil gives false positive with LOG.isWarning

### ISSUE THREAD

**Issue by tkleiber** [pmd/pmd#460]
**Rule Set:**
Java Logging: GuardLogStatementJavaUtil

**Description:**
The if outside LOG.fine is correctly identified, the second around LOG.warning not - but should.

**Code Sample demonstrating the issue:**

            if (LOG.isFine()) {
                LOG.fine("SUCCESS");
            }
            if (LOG.isWarning()) {
                LOG.warning("ERROR");
            }

**Running PMD through:** *[Ant]*

![if_log_jdeveloper](https://user-images.githubusercontent.com/2185441/27436105-7d9678fc-575e-11e7-9993-f015971122f3.png)
![if_log_jenkins](https://user-images.githubusercontent.com/2185441/27436106-7d97d274-575e-11e7-85ac-4ae145fb1fbd.png)

**Comment by adangel** [comment]
@tkleiber Which logging framework are you using? The method "isWarning()" doesn't exist for [java.util.logging.Logger](https://docs.oracle.com/javase/8/docs/api/java/util/logging/Logger.html).

We had in the past several issues, where this rule reported false positives on log4j and slf4j. This might be another one. As a workaround, it might help to disable this rule entirely for you.

**Comment by tkleiber** [comment]
We use in this case [ADFLogger](http://docs.oracle.com/cd/E84527_01/adf/api-reference-share/index.html?oracle/adf/share/logging/ADFLogger.html) from Oracle ADF.

So the first part with LOG.fine seem only does not analyzed in this case? Where is the difference?

**Comment by jsotuyod** [comment]
`ADFLogger` is not supported out of the box by any rules at the moment (although we probably will once we tackle #457 

In the meantime, however, you **can** configure the `GuardLogStatementJavaUtil` tool you are using to support it.

You can override the properties `guardsMethods` and `logLevels` as follows:

```xml
<property name="logLevels" value=".FINEST,.FINER,.FINE,.INFO,.INTERNAL,.CONFIG,.WARNING,.SEVERE" />
<property name="guardsMethods" value="isFinest,isFiner,isFine,isInfo,isInternal,isConfig,isWarning,isSevere" />
```

This however, will **not** detect properly logs guarded by `isLoggable(LogLevel)` calls.

**Comment by tkleiber** [comment]
Have now successfully configured the rule as described above, as we don't use  isLoggable calls at the moment.

**Comment by Fernal73** [comment]
Guard log statements have the potential to trigger the cyclomatic complexity rule.

Check out this post to read more:

It might be a good idea to incorporate some of the suggestions in the documentation of the rule.
